### PR TITLE
adding  mynn surface layer as opt_sfc=4  in NoahMP

### PR DESCRIPTION
**Pull request by HelinWei-NOAA**
## Description

This PR is to add MYNN surface scheme as opt_sfc=4 in NoahMP and recode the thermal roughness scheme as a subroutine. A bug with opt_sfc=1 was fixed. 2m T/Q from NoahMP instead of sfc_diag will be used as outputs.

The update is not aimed for UFS coupled prototype 8, but for RRFS.Is a change of answers expected from this PR?  Yes.

### Issue(s) addressed

ccpp-physics issue[ 920](https://github.com/NCAR/ccpp-physics/issues/920)

## Testing

How were these changes tested?  
What compilers / HPCs was it tested with?  
Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)  
Have the ufs-weather-model regression test been run? On what platform?  
- Will the code updates change regression test baseline? If yes, why? Please show the baseline directory below.
- Please commit the regression test log files in your ufs-weather-model branch

## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names (ideally)

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on ccpp-physics [PR 921](https://github.com/NCAR/ccpp-physics/pull/921)

### CODE REVIEW

**Comment by jkbk2004** [comment]
@DusanJovic-NOAA @junwang-noaa [ufs-wm pr 1224](https://github.com/ufs-community/ufs-weather-model/pull/1224) tests are done and ccpp dependency pr merged in. ready to merge in fv3atm side. @HelinWei-NOAA can you update ccpp pointers to make this pr ready to merge?

File: .gitmodules [hunk 1]
```diff
@@ -8,7 +8,7 @@
   branch = main
 [submodule "ccpp/physics"]
   path = ccpp/physics
-  url = https://github.com/NCAR/ccpp-physics
+  url = https://github.com/HelinWei-NOAA/ccpp-physics
```
Comment (junwang-noaa) [inline .gitmodules]: Helin, please change back to ncar ccpp physics repo.

**Comment by HelinWei-NOAA** [comment]
Yes. I updated gitmodules and ccpp-physics repository and my PR is ready to merge

File: .gitmodules (hunk 1 above)
Comment (HelinWei-NOAA) [reply .gitmodules]: done

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by junwang-noaa — APPROVED** [review]

**Pull request merged**